File: php/field_validation.php
// This script validates that certain fields are in the POST request when creating records.
// If not, an exception is thrown and 500 error returned to the client.
// If the script exits normally with no exception being thrown, then the records will be written to the database
// and the post-process script, if any, will be run.

// POST /api/v2/db/_table/account triggers script db._table.account.post.pre_process
// This script runs BEFORE records are written to the db.

$payload = $event['request']['payload'];

if(!empty($payload['resource'])){
    foreach($payload['resource'] as $record){
        if(!array_key_exists('annual_revenue', $record)){
            throw new \Exception('Missing field annual revenue.');
        }

        if($record['annual_revenue'] <= 0){
            throw new \Exception('Annual revenue must be > 0');
        }
    }
}

File: python3/field-validation.py
# POST /api/v2/db/_table/account triggers script db._table.account.post.pre_process
# This script runs BEFORE records are written to the db.

payload = event.request.payload

if hasattr(payload, 'resource') and payload.resource:
    for record in payload.resource:
        # Validate required fields
        if 'name' not in record:
            raise ValueError('Missing field: name')
        if 'email' not in record:
            raise ValueError('Missing field: email')
        
        # Validate field values
        if not record.name or record.name.strip() == '':
            raise ValueError('Name field cannot be empty')
        if not record.email or record.email.strip() == '':
            raise ValueError('Email field cannot be empty')
        
        # Optional: Validate email format
        if '@' not in record.email:
            raise ValueError('Invalid email format')
else:
    raise ValueError('No resource data found in payload')
